src/git.rs

```diff
@@ -78,65 +81,144 @@ impl Git {
 
-            let time_stamp = commit.time().seconds().to_string().parse::<i64>()?;
-            let date_time = DateTime::timestamp_2_utc(time_stamp);
-            self.update_str(COMMIT_DATE, date_time.human_format());
+        let time_stamp = commit.time().seconds().to_string().parse::<i64>()?;
+        let date_time = DateTime::timestamp_2_utc(time_stamp);
+        self.update_str(COMMIT_DATE, date_time.human_format());
 
-            self.update_str(COMMIT_DATE_2822, date_time.to_rfc2822());
+        self.update_str(COMMIT_DATE_2822, date_time.to_rfc2822());
 
-            self.update_str(COMMIT_DATE_3339, date_time.to_rfc3339());
+        self.update_str(COMMIT_DATE_3339, date_time.to_rfc3339());
 
-            let author = commit.author();
-            if let Some(v) = author.email() {
-                self.update_str(COMMIT_EMAIL, v.to_string());
-            }
+        let author = commit.author();
+        if let Some(v) = author.email() {
+            self.update_str(COMMIT_EMAIL, v.to_string());
+        }
 
-            if let Some(v) = author.name() {
-                self.update_str(COMMIT_AUTHOR, v.to_string());
-            }
-            let status_file = Self::git2_dirty_stage(&repo);
-            if status_file.trim().is_empty() {
-                self.update_bool(GIT_CLEAN, true);
-            } else {
-                self.update_bool(GIT_CLEAN, false);
-            }
-            self.update_str(GIT_STATUS_FILE, status_file);
+        if let Some(v) = author.name() {
+            self.update_str(COMMIT_AUTHOR, v.to_string());
         }
+        let status_file = Self::git2_dirty_stage(&repo);
+        if status_file.trim().is_empty() {
+            self.update_bool(GIT_CLEAN, true);
+        } else {
+            self.update_bool(GIT_CLEAN, false);
+        }
+        self.update_str(GIT_STATUS_FILE, status_file);
+
+        Ok(())
+    }
+
+    #[cfg(feature = "gitoxide")]
+    fn init_gitoxide(&mut self, path: &Path) -> SdResult<()> {
+        let repo = gix::discover(path).map_err(ShadowError::new)?;
+
+        let branch = repo
+            .head_name()
+            .map_err(ShadowError::new)?
+            .map(|n| n.shorten().to_string())
+            .unwrap_or_default();
+
+        // TODO: how does the --contains option actually work?
+        let refs = repo.references().map_err(ShadowError::new)?;
+        let tag = refs
+            .tags()
+            .map_err(ShadowError::new)?
+            .filter_map(|tag| match tag {
+                Ok(mut tag) => {
+                    let peeled_tag_id = match tag.inner.peeled {
+                        Some(id) => id,
+                        None => {
+                            tag.peel_to_id_in_place().ok()?;
+                            tag.inner.peeled.expect("peeled")
+                        }
+                    };
+                    repo.head_id().ok()?.eq(&peeled_tag_id).then_some(tag)
```

Comment:
`repo.head_id()` is expensive to obtain and should be initialized outside of the loop.